File: programming/javaClasses/SBST/buildBST.java
public class buildBST {
  static class Node {
    int data;
    Node left;
    Node right;
    Node(int data){
      this.data=data;
      this.left=null;
      this.right=null;
    }
    
  }
  public static Node insert(Node root,int data){
    if(root==null){
      root=new Node(data);
      return root;
    }
    if(root.data<data){
      root.right=insert(root.right,data);
    }
    else if(root.data> data){
      root.left=insert(root.left,data);
    }
    return root;
  }
  public static void inOrder(Node root){
    if(root==null){
      return ;
    }
    inOrder(root.left);
    System.out.print(root.data+" ");
    inOrder(root.right);
  }
  public static void main(String args[]){
    int values[]={10,11,12,1,3,4};
    Node root=null;
    for(int i=0;i<values.length;i++){
      root=insert(root,values[i]);
    }
    inOrder(root);
  }
}

File: programming/DS /c/binarySearchTrees/buildBST.c
#include<stdio.h>
#include<stdlib.h>
struct node{
    int data;
    struct node *left,*right;
};
struct node* newNode(int val){
    struct node* n=malloc(sizeof(struct node));
    n->data=val;
    n->left=NULL;
    n->right=NULL;
    return n;
}
struct node* buildBST(struct node* root,int val){
    if((root)==NULL){
        return newNode(val);
    }
    if(val<(root)->data){
        (root)->left=buildBST((root)->left,val);
    }
    if(val>(root)->data){
        (root)->right=buildBST((root)->right,val);
    }
    return root;

}
void inOrder(struct node* root){
    if(root==NULL){
        return;
    }
    inOrder(root->left);
    printf("%d ",root->data);
    inOrder(root->right);
} 
int main(){
    struct node* root=NULL;
    root=buildBST(root,5);
    buildBST(root,1);
    buildBST(root,7);
    buildBST(root,3);
    buildBST(root,2);
    buildBST(root,4);

    inOrder(root);
    printf("\n");
}